src/decorators/publicActionsChildChain.ts

```diff
@@ -0,0 +1,178 @@
+  SetParentPricingRewardRateParameters,
+  SetParentPricingRewardRateReturnType,
+} from '../actions/setParentPricingRewardRate';
+import {
+  setParentPricingRewardRecipient,
+  SetParentPricingRewardRecipientParameters,
+  SetParentPricingRewardRecipientReturnType,
+} from '../actions/setParentPricingRewardRecipient';
+import {
+  setSpeedLimit,
+  SetSpeedLimitParameters,
+  SetSpeedLimitReturnType,
+} from '../actions/setSpeedLimit';
+
+export type PublicActionsChildChain = {
+  // Getters
+  getAllChainOwners: (
+    parameters: GetAllChainOwnersParameters,
+  ) => Promise<GetAllChainOwnersReturnType>;
+  getInfraFeeAccount: (
+    parameters: GetInfraFeeAccountParameters,
+  ) => Promise<GetInfraFeeAccountReturnType>;
+  getNetworkFeeAccount: (
+    parameters: GetNetworkFeeAccountParameters,
+  ) => Promise<GetNetworkFeeAccountReturnType>;
+  getScheduledUpgrade: (
+    parameters: GetScheduledUpgradeParameters,
+  ) => Promise<GetScheduledUpgradeReturnType>;
+  isChainOwner: (parameters: IsChainOwnerParameters) => Promise<IsChainOwnerReturnType>;
+  getGasAccountingParams: (
+    parameters: GetGasAccountingParamsParameters,
+  ) => Promise<GetGasAccountingParamsReturnType>;
+  getMinimumGasPrice: (
+    parameters: GetMinimumGasPriceParameters,
+  ) => Promise<GetMinimumGasPriceReturnType>;
+  getParentBaseFeeEstimate: (
+    parameters: GetParentBaseFeeEstimateParameters,
+  ) => Promise<GetParentBaseFeeEstimateReturnType>;
+  getParentRewardRate: (
+    parameters: GetParentRewardRateParameters,
+  ) => Promise<GetParentRewardRateReturnType>;
+  getParentRewardRecipient: (
+    parameters: GetParentRewardRecipientParameters,
+  ) => Promise<GetParentRewardRecipientReturnType>;
+  // Setters
+  addChainOwner: (parameters: AddChainOwnerParameters) => Promise<AddChainOwnerReturnType>;
+  removeChainOwner: (parameters: RemoveChainOwnerParameters) => Promise<RemoveChainOwnerReturnType>;
+  setMaxTxGasLimit: (parameters: SetMaxTxGasLimitParameters) => Promise<SetMaxTxGasLimitReturnType>;
+  setParentPricePerUnit: (
+    parameters: SetParentPricePerUnitParameters,
+  ) => Promise<SetParentPricePerUnitReturnType>;
+  setParentPricingRewardRate: (
+    parameters: SetParentPricingRewardRateParameters,
+  ) => Promise<SetParentPricingRewardRateReturnType>;
+  setParentPricingRewardRecipient: (
+    parameters: SetParentPricingRewardRecipientParameters,
+  ) => Promise<SetParentPricingRewardRecipientReturnType>;
+  setSpeedLimit: (parameters: SetSpeedLimitParameters) => Promise<SetSpeedLimitReturnType>;
+};
+
+/**
+ * Public actions for child chain
+ *
+ * @example
+ * import { createPublicClient, http } from 'viem'
+ * import { publicActionsChildChain } from '@arbitrum/orbit-sdk'
+ *
+ * export const publicClientChildChain = createPublicClient({
+ *   chain: orbitChain,
+ *   transport: http(),
+ * }).extend(publicActionsChildChain())
+ *
+ * const isAChainOwner = await publicClientChildChain.isChainOwner({ address: zeroAddress })
```

Comment:
i know this is only an example but `isAChainOwner` deeply irritates me
can we change it to `isChainOwner` instead?